### ISSUE TITLE
Network Training

### ISSUE THREAD

**Issue by gancx** [ductuantruong/enskd#2]
Thanks for your efforts and work. I would like to know if you compute the loss between the ground truth and teacher logits and use it for error backpropagation. Will you also share the training log for reference?

**Comment by ductuantruong** [comment]
Hi @gancx,

Thank you for your interest in our work. For training, we only use 2 loss functions: classification loss (between student logits and the ground truth) and the knowledge distillation loss (between student and teacher logits). I also attached the training log showing loss values: [enskd_wavlm_campplus_emb512_g2.txt](https://github.com/ductuantruong/enskd/files/14569338/enskd_wavlm_campplus_emb512_g2.txt).

Hope this answers your concern. Please help me close the issues if it does.

**Comment by gancx** [comment]
Sorry to bother you. Actually, I am a little bit confused about the settings of the teacher model. Will you update the parameters of ecapa-tdnn that is on top of wavlm large pre-trained model? I found that only weights of wavlm are loaded, but the wavlm + ecapa-tdnn is set as no gradient flow.

**Comment by ductuantruong** [comment]
Hi @gancx,

The teacher model wavlm-ecapa-tdnn is frozen during training. The wavlm-ecapa-tdnn model has been already trained for speaker verification in the WavLM paper and the weight of the pretrained is for whole wavlm-ecapa-tdnn model (the weigh is provided by Microsoft as I included the link in the README). 

Hope this answers your concern.

**Comment by gancx** [comment]
I see. That makes sense to me. Thanks for your reply.
